### ISSUE TITLE
Can't save settings with XG3 on MacOSX

### ISSUE THREAD

**Issue by hrvstr** [lformella/xdcc-grabscher#33]
Hey I am using XG3 on Mac OSX 10.9.1 and I can't save any settings using the web interface. the save button is greyed out. Saving Filehandlers is greyed out as well.

```
localhost:~ HRVSTR$ mono --version
Mono JIT compiler version 3.2.3 ((no/8d3b4b7 Mon Sep 16 23:46:28 EDT 2013)
Copyright (C) 2002-2012 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
    TLS:           normal
    SIGSEGV:       altstack
    Notification:  kqueue
    Architecture:  x86
    Disabled:      none
    Misc:          softdebug 
    LLVM:          yes(3.3svn-mono)
    GC:            sgen
```

```
localhost:xg HRVSTR$ bash start.sh
localhost:xg HRVSTR$ /Users/HRVSTR/XG/Microsoft.AspNet.SignalR.Core.dll Error : 0 : SignalR exception thrown by Task: System.AggregateException: One or more errors occured ---> System.NotImplementedException: The requested feature is not implemented.
  at Microsoft.AspNet.SignalR.Owin.OwinWebSocketHandler+OwinWebSocket.get_State () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.WebSockets.WebSocketHandler.SendAsync (ArraySegment`1 message, WebSocketMessageType messageType, Boolean endOfMessage) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.WebSockets.DefaultWebSocketHandler.SendChunk (ArraySegment`1 message) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter.<.ctor>b__2 (ArraySegment`1 data, System.Object state) [0x00000] in <filename unknown>:0 
  at (wrapper delegate-invoke) System.Action`2<System.ArraySegment`1<byte>, object>:invoke_void__this___T1_T2 (System.ArraySegment`1<byte>,object)
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush (System.Byte[] byteBuffer, Boolean flushEncoder) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush (Boolean flushEncoder) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter.Flush () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Transports.WebSocketTransport.PerformSend (System.Object state) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Transports.WebSocketTransport.<Send>b__6 (System.Object state) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.TaskQueue.<Enqueue>b__1 (System.Func`2 next, System.Object nextState) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.TaskAsyncHelper.FromMethod[Func`2,Object] (System.Func`3 func, System.Func`2 arg1, System.Object arg2) [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[System.Boolean].GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.Subscription+<Invoke>d__2.MoveNext () [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[System.Boolean].GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.Subscription+<Work>d__8.MoveNext () [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.MessageBroker+<DoWork>d__2.MoveNext () [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
 --> (Inner exception 0) System.NotImplementedException: The requested feature is not implemented.
  at Microsoft.AspNet.SignalR.Owin.OwinWebSocketHandler+OwinWebSocket.get_State () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.WebSockets.WebSocketHandler.SendAsync (ArraySegment`1 message, WebSocketMessageType messageType, Boolean endOfMessage) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.WebSockets.DefaultWebSocketHandler.SendChunk (ArraySegment`1 message) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter.<.ctor>b__2 (ArraySegment`1 data, System.Object state) [0x00000] in <filename unknown>:0 
  at (wrapper delegate-invoke) System.Action`2<System.ArraySegment`1<byte>, object>:invoke_void__this___T1_T2 (System.ArraySegment`1<byte>,object)
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush (System.Byte[] byteBuffer, Boolean flushEncoder) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush (Boolean flushEncoder) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter.Flush () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Transports.WebSocketTransport.PerformSend (System.Object state) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Transports.WebSocketTransport.<Send>b__6 (System.Object state) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.TaskQueue.<Enqueue>b__1 (System.Func`2 next, System.Object nextState) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.TaskAsyncHelper.FromMethod[Func`2,Object] (System.Func`3 func, System.Func`2 arg1, System.Object arg2) [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[System.Boolean].GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.Subscription+<Invoke>d__2.MoveNext () [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[System.Boolean].GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.Subscription+<Work>d__8.MoveNext () [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.MessageBroker+<DoWork>d__2.MoveNext () [0x00000] in <filename unknown>:0 

/Users/HRVSTR/XG/Microsoft.AspNet.SignalR.Core.dll Error : 0 : SignalR exception thrown by Task: System.AggregateException: One or more errors occured ---> System.NotImplementedException: The requested feature is not implemented.
  at Microsoft.AspNet.SignalR.Owin.OwinWebSocketHandler+OwinWebSocket.get_State () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.WebSockets.WebSocketHandler.SendAsync (ArraySegment`1 message, WebSocketMessageType messageType, Boolean endOfMessage) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.WebSockets.DefaultWebSocketHandler.SendChunk (ArraySegment`1 message) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter.<.ctor>b__2 (ArraySegment`1 data, System.Object state) [0x00000] in <filename unknown>:0 
  at (wrapper delegate-invoke) System.Action`2<System.ArraySegment`1<byte>, object>:invoke_void__this___T1_T2 (System.ArraySegment`1<byte>,object)
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush (System.Byte[] byteBuffer, Boolean flushEncoder) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush (Boolean flushEncoder) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter.Flush () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Transports.WebSocketTransport.PerformSend (System.Object state) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Transports.WebSocketTransport.<Send>b__6 (System.Object state) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.TaskQueue.<Enqueue>b__1 (System.Func`2 next, System.Object nextState) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.TaskAsyncHelper+GenericDelegates`4+<>c__DisplayClass52[System.Object,System.Threading.Tasks.Task,System.Func`2[System.Object,System.Threading.Tasks.Task],System.Object].<ThenWithArgs>b__51 () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.TaskAsyncHelper+TaskRunners`2+<>c__DisplayClass3a[System.Object,System.Threading.Tasks.Task].<RunTask>b__39 (System.Threading.Tasks.Task t) [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[System.Boolean].GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.Subscription+<Invoke>d__2.MoveNext () [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[System.Boolean].GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.Subscription+<Work>d__8.MoveNext () [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.MessageBroker+<DoWork>d__2.MoveNext () [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
 --> (Inner exception 0) System.NotImplementedException: The requested feature is not implemented.
  at Microsoft.AspNet.SignalR.Owin.OwinWebSocketHandler+OwinWebSocket.get_State () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.WebSockets.WebSocketHandler.SendAsync (ArraySegment`1 message, WebSocketMessageType messageType, Boolean endOfMessage) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.WebSockets.DefaultWebSocketHandler.SendChunk (ArraySegment`1 message) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter.<.ctor>b__2 (ArraySegment`1 data, System.Object state) [0x00000] in <filename unknown>:0 
  at (wrapper delegate-invoke) System.Action`2<System.ArraySegment`1<byte>, object>:invoke_void__this___T1_T2 (System.ArraySegment`1<byte>,object)
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush (System.Byte[] byteBuffer, Boolean flushEncoder) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush (Boolean flushEncoder) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter+ChunkedWriter.Flush () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.BufferTextWriter.Flush () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Transports.WebSocketTransport.PerformSend (System.Object state) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Transports.WebSocketTransport.<Send>b__6 (System.Object state) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Infrastructure.TaskQueue.<Enqueue>b__1 (System.Func`2 next, System.Object nextState) [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.TaskAsyncHelper+GenericDelegates`4+<>c__DisplayClass52[System.Object,System.Threading.Tasks.Task,System.Func`2[System.Object,System.Threading.Tasks.Task],System.Object].<ThenWithArgs>b__51 () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.TaskAsyncHelper+TaskRunners`2+<>c__DisplayClass3a[System.Object,System.Threading.Tasks.Task].<RunTask>b__39 (System.Threading.Tasks.Task t) [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[System.Boolean].GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.Subscription+<Invoke>d__2.MoveNext () [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[System.Boolean].GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.Subscription+<Work>d__8.MoveNext () [0x00000] in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () [0x00000] in <filename unknown>:0 
  at Microsoft.AspNet.SignalR.Messaging.MessageBroker+<DoWork>d__2.MoveNext () [0x00000] in <filename unknown>:0 
```

**Comment by borkie** [comment]
Same under windows.
Workaround is ability "Elasticsearch" under config and add a port number, then disable it and now you can save it

**Comment by lformella** [comment]
Can you make a screenshot with the disabled button and your settings? If you do not want to post it here, send me an email to github{at}larsf{dot}.de

The save button is disabled If some of the fields have validation errors. If the fields are empty, they should be marked as valid.

**Comment by hrvstr** [comment]
I wasn't able to reproduce the error at first when I tried to today. I was able to save settings right after starting the server. After saving for the first time, the save button got greyed out again and i wasn't able to get it back working with the workaround from user borkie (add a elastic search port number and disable again).
Screenshot: http://i.imgur.com/BNqhXpw.png

I then stopped the server with the bash script, started it up again and the save button was still greyed out but I could get it to work with the workaround.
Screenshot: http://imgur.com/gAamhm3
